src/selectors/entities/channel_categories.ts

```diff
@@ -0,0 +1,375 @@
+        (categoryIds, categoriesById) => {
+            if (!categoryIds) {
+                return [];
+            }
+
+            return categoryIds.map((id) => categoriesById[id]);
+        }
+    );
+}
+
+export function makeGetUnsortedUnfilteredChannels(): (state: GlobalState, teamId: string) => Channel[] {
+    return createSelector(
+        (state: GlobalState) => state.entities.channels.channels,
+        getMyChannelMemberships,
+        (state: GlobalState, teamId: string) => teamId,
+        (allChannels: IDMappedObjects<Channel>, myMembers: RelationOneToOne<Channel, ChannelMembership>, teamId: string) => {
+            return Object.values(allChannels).
+                filter((channel) => channel.delete_at === 0).
+                filter((channel) => channel.team_id === teamId || channel.team_id === '').
+                filter((channel) => myMembers.hasOwnProperty(channel.id));
+        }
+    );
+}
+
+export function makeFilterChannelsByFavorites(): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
+    return createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
+        getMyPreferences,
+        (channels, categoryType, myPreferences) => {
+            const filtered = channels.filter((channel) => {
+                if (categoryType === CategoryTypes.FAVORITES) {
+                    return isFavoriteChannel(myPreferences, channel.id);
+                }
+
+                return !isFavoriteChannel(myPreferences, channel.id);
+            });
+
+            return filtered.length === channels.length ? channels : filtered;
+        }
+    );
+}
+
+export function makeFilterChannelsByType(): (state: GlobalState, channels: Channel[], categoryType: string) => Channel[] {
+    // This doesn't need to be a selector, but make it as one to keep it consistent
+    return createSelector(
+        (state: GlobalState, channels: Channel[]) => channels,
+        (state: GlobalState, channels: Channel[], categoryType: string) => categoryType,
+        (channels, categoryType) => {
+            const filtered = channels.filter((channel) => {
+                if (categoryType === CategoryTypes.PUBLIC) {
+                    return channel.type === General.OPEN_CHANNEL;
+                } else if (categoryType === CategoryTypes.PRIVATE) {
+                    return channel.type === General.PRIVATE_CHANNEL;
+                } else if (categoryType === CategoryTypes.DIRECT_MESSAGES) {
+                    return channel.type === General.DM_CHANNEL || channel.type === General.GM_CHANNEL;
+                }
+
+                return true;
+            });
+
+            return filtered.length === channels.length ? channels : filtered;
+        }
+    );
+}
+
+function getDefaultAutocloseCutoff() {
+    return Date.now() - 7 * 24 * 60 * 60 * 1000;
```

Comment:
Non-blocking, but maybe the 7 days should be a constant somewhere?

Reply:
That would certainly be helpful, but I'd say it's out of scope of this PR. Maybe `moment` provides some sort of constants for this